src/sentry/services/hybrid_cloud/organization/impl.py

```diff
@@ -518,6 +520,18 @@ def send_signal(
     ) -> None:
         signal.signal.send_robust(None, organization_id=organization_id, **args)
 
+    def get_teams_for_user(self, *, organization_id: int, user_id: int) -> List[RpcTeam]:
+        from sentry.services.hybrid_cloud.user.service import user_service
+
+        organization = Organization.objects.get_from_cache(id=organization_id)
+        user = user_service.get_user(user_id=user_id)
```

Comment:
This seems unwise.  Taking a peak at `get_for_user` on the Team manager it looks like the *only* thing it needs from user is the id and `is_authenticated` (which doesn't matter in our case).

Rather than increase the amount of back and forth RPC calls here, why don't we change the `get_for_user` that might just take a user_id directly and skip needing this lookup at all?

Reply:
@corps  I've updated and refactored `Team.objects.get_for_user()`